src/test/java/com/artipie/docker/DockerProxyTest.java

```diff
@@ -0,0 +1,128 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.docker;
+
+import com.amihaiemil.eoyaml.Yaml;
+import com.artipie.RepoConfig;
+import com.artipie.asto.Key;
+import com.artipie.http.Headers;
+import com.artipie.http.Slice;
+import com.artipie.http.hm.RsHasStatus;
+import com.artipie.http.rq.RequestLine;
+import com.artipie.http.rq.RqMethod;
+import com.artipie.http.rs.RsStatus;
+import io.reactivex.Flowable;
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+import org.eclipse.jetty.client.HttpClient;
+import org.hamcrest.MatcherAssert;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Tests for {@link DockerProxy}.
+ *
+ * @since 0.9
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+class DockerProxyTest {
+
+    @Test
+    void shouldBuildFromConfig() throws Exception {
+        final Slice slice = dockerProxy(
+            String.join(
+                "\n",
+                "repo:",
+                "  type: docker-proxy",
+                "  remotes:",
+                "    - url: registry-1.docker.io",
+                "      username: admin",
+                "      password: qwerty",
+                "      cache:",
+                "        storage:",
+                "          type: fs",
+                "          path: /var/artipie/data/cache",
+                "    - url: another-registry.org:54321",
+                "    - url: mcr.microsoft.com",
+                "      cache:",
+                "        storage: my-storage",
+                "  storage:",
+                "    type: fs",
+                "    path: /var/artipie/data/local"
+            )
+        );
+        MatcherAssert.assertThat(
+            slice.response(
+                new RequestLine(RqMethod.GET, "/v2/").toString(),
+                Headers.EMPTY,
+                Flowable.empty()
+            ),
+            new RsHasStatus(RsStatus.OK)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("badConfigs")
```

Comment:
@olegmoz I'd suggest to unify this test: either use `MethodSource` in both test methods or nowhere

Reply:
@olenagerasimova thanks, updated for uniformity